src/common/components/vanilla/button/index.js

```diff
@@ -1,53 +1,58 @@
 import React, { PropTypes } from 'react';
 import { Link } from 'react-router';
+import classNames from 'classnames';
 
 import Spinner from '../../spinner';
 import style from './button.css';
 
-const defaultProps = {
-  isSpinner: PropTypes.bool,
-  disabled: PropTypes.bool,
-  children: PropTypes.string,
-  onClick: PropTypes.func,
-  type: PropTypes.string,
-  appearance: React.PropTypes.oneOf(['positive', 'negative', 'neutral', 'base']),
-  flavour: React.PropTypes.oneOf(['normal','bigger', 'smaller']),
-  href: PropTypes.string
-};
-
-export default function Button(props) {
-  const { appearance='primary', flavour='normal', isSpinner=false, ...rest } = props;
-  return (
-    <button {...rest} className={ `${style[appearance]} ${style[flavour]}` }>
-      { isSpinner && <span className={ style.spinner }><Spinner /></span> }
-      { props.children }
-    </button>
-  );
+function createButtonComponent(Component) {
+  function ButtonComponent(props) {
```

Comment:
We have 3 button classes (`Button`, `Anchor` and `LinkButton` - for `<button>`, `<a>`, and React router `Link`). Previously they had separate implementations that were meant to share functionality, props, etc, but in the end each of them was different (supporting different subset of props).

This refactoring fixes that. `createButtonComponent` is a factory function that builds a button component for given component (or tag name).